bundles/org.openhab.core/src/main/java/org/openhab/core/library/unit/CurrencyUnit.java

```diff
@@ -0,0 +1,287 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.library.unit;
+
+import static org.eclipse.jdt.annotation.DefaultLocation.FIELD;
+import static org.eclipse.jdt.annotation.DefaultLocation.PARAMETER;
+import static org.eclipse.jdt.annotation.DefaultLocation.RETURN_TYPE;
+import static org.eclipse.jdt.annotation.DefaultLocation.TYPE_BOUND;
+import static org.openhab.core.library.unit.CurrencyUnits.SYSTEM_CURRENCY;
+import static tech.units.indriya.AbstractUnit.ONE;
+
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.measure.Dimension;
+import javax.measure.IncommensurableException;
+import javax.measure.Prefix;
+import javax.measure.Quantity;
+import javax.measure.UnconvertibleException;
+import javax.measure.Unit;
+import javax.measure.UnitConverter;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.internal.library.unit.CurrencyConverter;
+import org.openhab.core.internal.library.unit.CurrencyService;
+import org.openhab.core.library.dimension.Currency;
+
+import tech.units.indriya.format.SimpleUnitFormat;
+import tech.units.indriya.function.AbstractConverter;
+import tech.units.indriya.function.AddConverter;
+import tech.units.indriya.function.Calculus;
+import tech.units.indriya.function.MultiplyConverter;
+import tech.units.indriya.function.RationalNumber;
+import tech.units.indriya.internal.function.Calculator;
+import tech.units.indriya.unit.AlternateUnit;
+import tech.units.indriya.unit.ProductUnit;
+import tech.units.indriya.unit.TransformedUnit;
+import tech.units.indriya.unit.UnitDimension;
+import tech.uom.lib.common.function.Nameable;
+import tech.uom.lib.common.function.PrefixOperator;
+import tech.uom.lib.common.function.SymbolSupplier;
+
+/**
+ * The {@link CurrencyUnit} is a
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault({ PARAMETER, RETURN_TYPE, FIELD, TYPE_BOUND })
+public final class CurrencyUnit implements Unit<Currency>, Comparable<Unit<Currency>>, PrefixOperator<Currency>,
+        Nameable, Serializable, SymbolSupplier {
+
+    private static final long serialVersionUID = -1L;
+    private static final Dimension DIMENSION = UnitDimension.parse('$');
+    private String name;
+    private String symbol;
```

Comment:
I assume name is "Euro" and symbol is "€", as an example? Should we also add currency code ("EUR"), which seems more important than the symbol? And probably symbol should be optional since I believe not all currencies have one.